products/dns/api.yaml

```diff
@@ -91,6 +95,35 @@ objects:
           A set of key/value label pairs to assign to this ManagedZone.
         update_verb: :PATCH
         update_url: 'projects/{{project}}/managedZones/{{name}}'
+      - !ruby/object:Api::Type::Enum
+        name: 'visibility'
+        description: |
+          Visibility of the zone. Public zones are visible to the public
+          internet. Private zones are only visible in your internal networks
+          denoted by the `networks` argument. Must be one of: `public`,
+          `private`.
+        values:
+        - :private
+        - :public
+        default_value: :public
+        min_version: beta
+      - !ruby/object:Api::Type::NestedObject
+        name: 'privateVisibilityConfig'
+        description: 'Visibility config for private zones.'
+        update_verb: :PATCH
+        update_url: 'projects/{{project}}/managedZones/{{name}}'
+        properties:
+          - !ruby/object:Api::Type::Array
+            name: 'networks'
+            description: 'List of networks.'
+            item_type: !ruby/object:Api::Type::NestedObject
+              properties:
+                # TODO(drebes): Make 'networkUrl' a ResourceRef once cross-module references
```

Comment:
Since it isn't a ResourceRef, you'll probably want to add some of the special things we do for URLs (`diff_suppress_func: 'compareSelfLinkOrResourceName'` in the ResourceOverride for tf should be sufficient)